src/main/java/billgates/view/gui/ActionPanel.java

```diff
@@ -0,0 +1,160 @@
+    private final JPasswordField passwordField = new JPasswordField(16);
+
+    private final JButton signInButton = new ActionButton("Sign In");
+    private final JButton signOutButton = new ActionButton("Sign Out");
+    private final JButton backButton = new ActionButton(" Back", backIcon);
+    private final JButton addEntryButton = new ActionButton("Add Entry");
+    private final JButton deleteEntryButton = new ActionButton("Delete Entry");
+    private final JTextArea statisticsTextArea = new JTextArea("Statistics");
+
+    // private Icon backIcon;
+
+    public ActionPanel() {
+        this.setLayout(this.layout);
+        this.initSignInPanel();
+        this.initButtonTextArea();
+        this.setPreferredSize(new Dimension(DEFAULT_WIDTH, DEFAULT_HEIGHT));
+        this.initBorder();
+    }
+
+    private void initSignInPanel() {
+        this.add(this.signInPanel);
+        // Set the size of signInPanel
+        this.signInPanel.setMaximumSize(new Dimension(DEFAULT_SIGN_IN_PANEL_WIDTH, DEFAULT_SIGN_IN_PANEL_HEIGHT));
+
+        // add and layout components
+        // username label
+        this.signInPanel.add(this.usernameLabel);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.usernameLabel, HORIZONTAL_GAP,
+                SpringLayout.WEST, this.signInPanel);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.usernameLabel, VERTICAL_GAP,
+                SpringLayout.NORTH, this.signInPanel);
+
+        // username field
+        this.signInPanel.add(this.usernameField);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.usernameField, HORIZONTAL_GAP,
+                SpringLayout.EAST, this.usernameLabel);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.usernameField, VERTICAL_GAP,
+                SpringLayout.NORTH, this.signInPanel);
+
+        // password label
+        this.signInPanel.add(this.passwordLabel);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.passwordLabel, HORIZONTAL_GAP,
+                SpringLayout.WEST, this.signInPanel);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.passwordLabel, VERTICAL_GAP,
+                SpringLayout.SOUTH, this.usernameLabel);
+
+        // password field
+        this.signInPanel.add(this.passwordField);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.passwordField, 0,
+                SpringLayout.WEST, this.usernameField);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.passwordField, VERTICAL_GAP,
+                SpringLayout.SOUTH, this.usernameField);
+    }
+
+    private void initButtonTextArea(){
+        // Components with gaps
+        // signInButton
+        this.add(signInButton);
+        signInButton.setAlignmentX(CENTER_ALIGNMENT);
+        this.add(Box.createRigidArea(new Dimension(0,VERTICAL_GAP)));
+        // Sign in event
+        signInButton.addActionListener((e -> signIn()));
```

Comment:
For all instance methods, a better practice is to add `this.` before the instance methods.